solidity/test/bridge/Bridge.Frauds.test.ts

```diff
@@ -0,0 +1,1528 @@
+                  challengeKey
+                )
+
+                expect(fraudChallenge.challenger).to.equal(
+                  await thirdParty.getAddress()
+                )
+                expect(fraudChallenge.depositAmount).to.equal(
+                  await bridge.fraudChallengeDepositAmount()
+                )
+                expect(fraudChallenge.reportedAt).to.equal(
+                  await lastBlockTime()
+                )
+                expect(fraudChallenge.closed).to.equal(false)
+              })
+
+              it("should emit FraudChallengeSubmitted event", async () => {
+                await expect(tx)
+                  .to.emit(bridge, "FraudChallengeSubmitted")
+                  .withArgs(
+                    walletPublicKeyHash,
+                    data.sighash,
+                    data.signature.v,
+                    data.signature.r,
+                    data.signature.s
+                  )
+              })
+            })
+
+            context("when the fraud challenge already exists", () => {
+              before(async () => {
+                await createSnapshot()
+
+                await bridge.setWallet(walletPublicKeyHash, {
+                  ecdsaWalletID: ethers.constants.HashZero,
+                  mainUtxoHash: ethers.constants.HashZero,
+                  pendingRedemptionsValue: 0,
+                  createdAt: await lastBlockTime(),
+                  state: walletState.Live,
+                })
+
+                await bridge
+                  .connect(thirdParty)
+                  .submitFraudChallenge(
+                    walletPublicKey,
+                    data.sighash,
+                    data.signature,
+                    {
+                      value: await bridge.fraudChallengeDepositAmount(),
+                    }
+                  )
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              it("should revert", async () => {
+                await expect(
+                  bridge
+                    .connect(thirdParty)
+                    .submitFraudChallenge(
+                      walletPublicKey,
+                      data.sighash,
+                      data.signature,
+                      {
+                        value: await bridge.fraudChallengeDepositAmount(),
+                      }
+                    )
+                ).to.be.revertedWith("Fraud challenge already exists")
+              })
+            })
+          }
+        )
+
+        context("when incorrect wallet public key is used", () => {
+          const incorrectWalletPublicKey =
+            "0xffc045ade19f8a5d464299146ce069049cdcc2390a9b44d9abcd83f11d8cce4" +
+            "01ea6800e307b87aadebdcd2f7293cc60f0526afaff1a7b1abddfd787e6c5871e"
+
+          const incorrectWalletPublicKeyHash =
+            "0xb5222794425b9b8cd8c3358e73a50dea73480927"
```

Comment:
Worth dropping a comment explaining the incorrectness of that key. It's not easy to grasp what exactly causes the revert here. Same for all `when incorrect*` scenarios below.

Reply:
Done in 8e8feed7b7b55753b47ade47d2ac82db25e77519.